redis-om-spring/src/main/java/com/redis/om/spring/search/stream/actions/BaseAbstractAction.java

```diff
@@ -14,11 +14,7 @@ public abstract class BaseAbstractAction implements TakesJSONOperations {
   public BaseAbstractAction(Field field) {
     this.field = field;
     Optional<Field> maybeId = ObjectUtils.getIdFieldForEntityClass(field.getDeclaringClass());
-    if (maybeId.isPresent()) {
-      this.idField = maybeId.get();
-    } else {
-      throw new IllegalArgumentException(field.getDeclaringClass().getName() + " does not appear to have an ID field");
-    }
+    this.idField = maybeId.get();
```

Comment:
if we're removign the check - do we really need this to be an Optional?

Reply:
The check was unnecessary, classes without an Id do not get this far! The annotation processor fails before they get here.